homeassistant/components/nexia/binary_sensor.py

```diff
@@ -0,0 +1,114 @@
+"""Support for Nexia / Trane XL Thermostats."""
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.const import ATTR_ATTRIBUTION
+
+from .const import (
+    ATTR_THERMOSTAT_ID,
+    ATTRIBUTION,
+    DATA_NEXIA,
+    DOMAIN,
+    MANUFACTURER,
+    NEXIA_DEVICE,
+    UPDATE_COORDINATOR,
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up sensors for a Nexia device."""
+
+    nexia_data = hass.data[DOMAIN][config_entry.entry_id][DATA_NEXIA]
+    nexia_home = nexia_data[NEXIA_DEVICE]
+    coordinator = nexia_data[UPDATE_COORDINATOR]
+
+    entities = []
+    for thermostat_id in nexia_home.get_thermostat_ids():
+        thermostat = nexia_home.get_thermostat_by_id(thermostat_id)
+        entities.append(
+            NexiaBinarySensor(
+                coordinator, thermostat, "is_blower_active", "Blower Active", None
+            )
+        )
+        if thermostat.has_emergency_heat():
+            entities.append(
+                NexiaBinarySensor(
+                    coordinator,
+                    thermostat,
+                    "is_emergency_heat_active",
+                    "Emergency Heat Active",
+                    None,
+                )
+            )
+
+    async_add_entities(entities, True)
+
+
+class NexiaBinarySensor(BinarySensorDevice):
+    """Provices Nexia BinarySensor support."""
+
+    def __init__(self, coordinator, device, sensor_call, sensor_name, sensor_class):
+        """Initialize the nexia sensor."""
+        self._coordinator = coordinator
+        self._device = device
+        self._name = self._device.get_name() + " " + sensor_name
```

Comment:
Here you calculate it once, I also saw in another entity that you calculate it inside the property.

Reply:
Fixed